Added OpenLDAP chart

enis opened the pull request:
An initial helm chart for installing OpenLDAP server. 
It uses docker images from https://github.com/osixia/docker-openldap. Comes with some documentation and basic test. The chart is added to the incubator until it matures.

enis committed:
Added OpenLDAP chart

viglesiasce commented:
@enis can you check the CLA?

enis commented:
/retest

enis commented:
@viglesiasce thanks for checking. I've finally resolved the CCLA problems. It took some time.

unguiculus reviewed (changes requested):
* Please make sure you have standard labels on all resouces. 
* The `app` label should be `{{ template "openldap.name" . }}`.
* Please update `_helpers.tpl` to the one created with `helm create` using Helm 2.8+.

unguiculus commented:
ping @enis Any change you can update the PR?

enis committed:
Rebased the code. Addressed review items and tried to bring the chart to be more consistent to other charts in the repo

enis commented:
Sorry this fell from my radar. I have rebased the patch, and addressed the review comments. @unguiculus  do you mind taking a look. Thanks.

enis commented:
/assign @unguiculus

unguiculus commented on incubator/openldap/templates/openldap-customldif-configmap.yaml:
```diff
@@ -0,0 +1,20 @@
+  name: {{ template "openldap.fullname" . }}-customldif-configmap
+  labels:
+    app: {{ template "openldap.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
`chart: {{ template "openldap.chart" . }}`

Apply everywhere.

unguiculus commented on incubator/openldap/templates/openldap-deployment.yaml:
```diff
@@ -0,0 +1,78 @@
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        ports:
+        - name: ldap
+          containerPort: {{ .Values.service.ldapInternalPort }}
```
It probably doesn't make sense to make container ports configurable. Change them will break things. I'd suggest you hard-code them, give them names, and reference them via their names from service and probes (which you should add).

unguiculus commented on incubator/openldap/templates/openldap-pvc.yaml:
```diff
@@ -0,0 +1,27 @@
+    heritage: {{ .Release.Service }}
+spec:
+  accessModes:
+    - {{ .Values.persistence.accessMode | quote }}
```
I'd like you to make list indentation consistent throughout the chart. Either use zero or two spaces.

enis committed:
Addressed review comments:
  - Reindented charts with 2 spaces for list items
  - Hard coded container port name to refer to it in the service
  - Added liveness and readiness probes
  - Changed chart label values to use the template

unguiculus commented on incubator/openldap/templates/openldap-service.yaml:
```diff
@@ -0,0 +1,26 @@
+#
+# Service spec for openldap
+#
```
Most of these comments are useless and just state the obvious. I'd remove them.

enis replied:
Removed comments.

unguiculus commented on incubator/openldap/templates/tests/openldap-test-runner.yaml:
```diff
@@ -0,0 +1,40 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ template "openldap.fullname" . }}-service-test-{{ randAlphaNum 5 | lower }}"
```
Add standard labels.

enis replied:
done.

unguiculus commented on incubator/openldap/templates/tests/openldap-tests.yaml:
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "openldap.fullname" . }}-tests
```
Add standard labels.

enis replied:
done.

unguiculus commented on incubator/openldap/templates/openldap-deployment.yaml:
```diff
@@ -0,0 +1,88 @@
+#
+# A Deployment spec for openldap slapd
+#
+apiVersion: apps/v1beta1
```
I'd suggest you use `apps/v1beta2`

enis replied:
done.

unguiculus commented on incubator/openldap/README.md:
```diff
@@ -0,0 +1,80 @@
+# OpenLDAP Helm Chart
+
+## Prerequisites Details
+* Kubernetes 1.6+
```
1.8+

enis replied:
fixed.

unguiculus reviewed (changes requested):
Next time, please merge instead of rebasing, so Github can display the changes. I'd suggest you move the chart to stable.

enis committed:
Moved the chart to stable, and addressed rest of the review feedback

enis commented:
Thanks @unguiculus I have addressed the review items.

enis committed:
Move to apps/v1beta2

unguiculus commented on stable/openldap/README.md:
```diff
@@ -0,0 +1,80 @@
+The following tables lists the configurable parameters of the openldap chart and their default values.
+
+| Parameter                       | Description                           | Default                                                    |
+| ------------------------------- | ----------------------------------    | ---------------------------------------------------------- |
```
Please format the table properly (align the pipes).

unguiculus commented:
/ok-to-test

enis committed:
Fixed formatting of markdown table in README

enis committed:
Fixed default value in README

unguiculus commented on stable/openldap/values.yaml:
```diff
@@ -0,0 +1,72 @@
+
+# Default Passwords to use. Unset these to auto-generate password stored as secrets.
+# You can override these at install time with
+# helm install openldap --set openldap.adminPassword=<passwd>,openldap.configPassword=<passwd>
```
Don't set default passwords. Default should be auto-generation.

enis replied:
fixed.

enis committed:
Auto-generate the passwords by default

unguiculus commented:
/lgtm
